cypress/integration/AboutMeAndLocation.feature

```diff
@@ -0,0 +1,49 @@
+Feature: About me and and location
+  As a user
+  I would like to add some about me text and a location
+  So others can get some info about me and my location
+
+  The location and about me are show on the user. Later it will be possible
+  to search for users by location.
+
+  Background:
+    Given I am logged in
+
+  Scenario: I change my name
+    When I visit the "settings" page
+    And I enter "Hansi" as my name
+    And I press "Save"
+    And I can click on my profile picture in the top right corner
+    Then I can see a "Hansi" as my name
+
+    When I refresh the page
+    And I can click on my profile picture in the top right corner
+    Then I can see a "Hansi" as my name
```

Comment:
```gherkin
    When I refresh the page
    Then my new name is still there
```

Reply:
How can I get the value of the name variable in a generic method? I can`t so far so I need to pass it again.

Reply:
There is absolutely no problem in sharing state between step definitions. So, your step definition `Then my new name is still there` could have the name "Hansi" hard coded. Is this a problem? No. As a developer you're trained to avoid hard-coded strings. But this is different. Your step is only the documentation for your implementation. In this case, it is more human-readable not to have "Hansi" in the step. Re-usability of step definition is not really important. Cucumber steps are not only method calls - they are the documentation and that's their main objective.

Verbose step definitions are a never ending story. Here are more examples by Aslak Hellesøy, the inventor of cucumber-testing: http://aslakhellesoy.com/post/11055981222/the-training-wheels-came-off

Reply:
The whole thing can be summed up with "Specification by example" and "Executable documentation".